src/onyx/plugin/kafka.clj

```diff
@@ -178,19 +197,30 @@
       (let [rec ^ConsumerRecord (.next ^java.util.Iterator iter)
             deserialized (some-> rec segment-fn)]
         (.set watermark (max (.get watermark) (.timestamp rec)))
-        (cond (= :done deserialized)
-              (do (set! drained true)
-                  nil)
+
+        (cond (= :done deserialized) ;; TODO: Remove this in favor of target-offsets
```

Comment:
Address now or later?

Reply:
This would require switching the tests over to remove `:done`. Up to you whether you want to do it now.

Reply:
Ok, pass for now. Circle back soon.